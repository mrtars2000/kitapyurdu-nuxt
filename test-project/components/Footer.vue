<template>
  <v-card height="600px" width="100%">
    <v-footer
      v-bind="localAttrs"
      :padless="padless"
      class="lighten-1"
    >
      <v-card
        flat
        tile
        width="100%"
        class="lighten-1 "
      >
        <v-card-text>
          <v-row class="space-between">
            <v-col>
              Mobil Uygulamalar
              <v-btn
                style="background-color: black;"
                class="white--text mx-4"
                x-small
              >
                App Store'dan
              </v-btn>
              <v-btn
                style="background-color: black;"
                class="white--text mx-4"
                x-small
              >
                Google Play
              </v-btn>
              <v-btn
                style="background-color: black;"
                class="white--text mx-4"
                x-small
              >
                AppGalery
              </v-btn>
            </v-col>

            <v-col class="text-right colors">
              Sosyal Medya:
              <v-btn
                v-for="icon in icons"
                :key="icon"
                class="mx-4"
                icon
                color="primary"
              >
                <v-icon size="24px">
                  {{ icon }}
                </v-icon>
              </v-btn>
            </v-col>
          </v-row>
        </v-card-text>
      </v-card>
    </v-footer>

    <v-row class="text-center" style="margin-top: 10px;">
      <v-col>
        Yardımcı Sayfalar:
        <v-col class="font">
          Yardım ve İşlem Rehberi
        </v-col>
        <v-col class="font">
          Video Yardım
        </v-col>
        <v-col class="font">
          Havale/EFT Bilgileri
        </v-col>
        <v-col class="font">
          Müşteri Hizmetleri
        </v-col>
        <v-col class="font">
          Ürün İadesi
        </v-col>
      </v-col>
      <v-col>
        Popüler Sayfalar:
        <v-col class="font">
          Çok Satan Kitaplar
        </v-col>
        <v-col class="font">
          Yeni Çıkan Kitaplar
        </v-col>
        <v-col class="font">
          Kargo Ödeyen Avantajlı Ürünler
        </v-col>
        <v-col class="font">
          Kampanyalar
        </v-col>
        <v-col class="font">
          Ayın Yazarları
        </v-col>
        <v-col class="font">
          Ayın Yayınevleri
        </v-col>
        <v-col class="font">
          Puan Kataloğu
        </v-col>
        <v-col class="font">
          E-Dergi
        </v-col>
        <v-col class="font">
          Yazarlar Ne okuyor?
        </v-col>
        <v-col class="font">
          Ne okusam
        </v-col>
      </v-col>
      <v-col>
        Özel Sayfalar:
        <v-col class="font">
          Sınav Kitapları
        </v-col>
        <v-col class="font">
          Çocuk Kitapları
        </v-col>
        <v-col class="font">
          Cep Boy Kitaplar
        </v-col>
        <v-col class="font">
          İngilizce Kitaplar
        </v-col>
        <v-col class="font">
          Lisanslı Ürünler
        </v-col>
        <v-col class="font">
          Bookinzi Kültür-Sanat Ürünleri
        </v-col>
        <v-col class="font">
          Oyundan Daha Fazlası
        </v-col>
        <v-col class="font">
          Montessori Oyuncakları
        </v-col>
        <v-col class="font">
          Ahşap Puzzle
        </v-col>
        <v-col class="font">
          Akıl Defteri
        </v-col>
      </v-col>
      <v-col>
        Kitapyurdu Platformları:
        <v-col class="font">
          Kitapyurdu Sesli Kitap
        </v-col>
        <v-col class="font">
          KDY Platformu
        </v-col>
        <v-col class="font">
          KDD Platformu
        </v-col>
        <v-col class="font">
          Kütüphanem Uygulaması
        </v-col>
        <v-col class="font">
          Kitap Kurdu Eklentisi
        </v-col>
      </v-col>
    </v-row>
  </v-card>
</template>

<script>
export default {
  data: () => ({
    icons: [
      'mdi-facebook',
      'mdi-twitter',
      'mdi-youtube',
      'mdi-instagram'
    ],
    items: [
      'default',
      'absolute',
      'fixed'
    ],
    colors: [
      'blue',
      'blue lighten-2',
      'red',
      'red lighten-2'
    ],
    padless: false,
    variant: 'default'
  }),
  computed: {
    localAttrs () {
      const attrs = {}

      if (this.variant === 'default') {
        attrs.absolute = false
        attrs.fixed = false
      } else {
        attrs[this.variant] = true
      }
      return attrs
    }
  }
}
</script>

<style>
    .font{
        font-weight: lighter;
    }
    .colors :nth-child(1) v-btn{
        color: blue;
    }
</style>
