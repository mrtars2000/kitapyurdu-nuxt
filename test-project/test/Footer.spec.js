import { shallowMount } from '@vue/test-utils'
import Vue from 'vue'
import Vuetify from 'vuetify'
import Vuex from 'vuex'
import Footer from '@/components/Footer'
Vue.use(Vuetify)
Vue.use(Vuex)

const factory = () => {
  return shallowMount(Footer, {})
}

describe('Footer.vue', () => {
  // eslint-disable-next-line no-unused-vars
  let vuetify
  let store

  beforeEach(() => {
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    vuetify = new Vuetify()

    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    store = new Vuex.Store({
      data: () => ({
        icons: [
          'mdi-facebook',
          'mdi-twitter',
          'mdi-youtube',
          'mdi-instagram'
        ],
        items: [
          'default',
          'absolute',
          'fixed'
        ],
        colors: [
          'blue',
          'blue lighten-2',
          'red',
          'red lighten-2'
        ],
        padless: false,
        variant: 'default'
      }),
      computed: {
        localAttrs () {
          const attrs = {}

          if (this.variant === 'default') {
            attrs.absolute = false
            attrs.fixed = false
          } else {
            attrs[this.variant] = true
          }
          return attrs
        }
      }
    })
  })

  const wrapper = factory()

  it('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('Snapshot property ', () => {
    const wrapper = factory()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
