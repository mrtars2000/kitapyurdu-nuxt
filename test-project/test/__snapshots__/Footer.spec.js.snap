// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer.vue Snapshot property  1`] = `
"<v-card-stub loaderheight=\\"4\\" tag=\\"div\\" height=\\"600px\\" width=\\"100%\\">
  <v-footer-stub height=\\"auto\\" tag=\\"footer\\" class=\\"lighten-1\\">
    <v-card-stub loaderheight=\\"4\\" tag=\\"div\\" width=\\"100%\\" tile=\\"true\\" flat=\\"true\\" class=\\"lighten-1 \\">
      <v-card-text-stub tag=\\"div\\">
        <v-row-stub tag=\\"div\\" class=\\"space-between\\">
          <v-col-stub tag=\\"div\\">
            Mobil Uygulamalar
            <v-btn-stub tag=\\"button\\" activeclass=\\"\\" xsmall=\\"true\\" type=\\"button\\" class=\\"white--text mx-4\\" style=\\"background-color: black;\\">
              App Store'dan
            </v-btn-stub>
            <v-btn-stub tag=\\"button\\" activeclass=\\"\\" xsmall=\\"true\\" type=\\"button\\" class=\\"white--text mx-4\\" style=\\"background-color: black;\\">
              Google Play
            </v-btn-stub>
            <v-btn-stub tag=\\"button\\" activeclass=\\"\\" xsmall=\\"true\\" type=\\"button\\" class=\\"white--text mx-4\\" style=\\"background-color: black;\\">
              AppGalery
            </v-btn-stub>
          </v-col-stub>
          <v-col-stub tag=\\"div\\" class=\\"text-right colors\\">
            Sosyal Medya:
            <v-btn-stub color=\\"primary\\" tag=\\"button\\" activeclass=\\"\\" icon=\\"true\\" type=\\"button\\" class=\\"mx-4\\">
              <v-icon-stub size=\\"24px\\">
                mdi-facebook
              </v-icon-stub>
            </v-btn-stub>
            <v-btn-stub color=\\"primary\\" tag=\\"button\\" activeclass=\\"\\" icon=\\"true\\" type=\\"button\\" class=\\"mx-4\\">
              <v-icon-stub size=\\"24px\\">
                mdi-twitter
              </v-icon-stub>
            </v-btn-stub>
            <v-btn-stub color=\\"primary\\" tag=\\"button\\" activeclass=\\"\\" icon=\\"true\\" type=\\"button\\" class=\\"mx-4\\">
              <v-icon-stub size=\\"24px\\">
                mdi-youtube
              </v-icon-stub>
            </v-btn-stub>
            <v-btn-stub color=\\"primary\\" tag=\\"button\\" activeclass=\\"\\" icon=\\"true\\" type=\\"button\\" class=\\"mx-4\\">
              <v-icon-stub size=\\"24px\\">
                mdi-instagram
              </v-icon-stub>
            </v-btn-stub>
          </v-col-stub>
        </v-row-stub>
      </v-card-text-stub>
    </v-card-stub>
  </v-footer-stub>
  <v-row-stub tag=\\"div\\" class=\\"text-center\\" style=\\"margin-top: 10px;\\">
    <v-col-stub tag=\\"div\\">
      Yardımcı Sayfalar:
      <v-col-stub tag=\\"div\\" class=\\"font\\">
        Yardım ve İşlem Rehberi
      </v-col-stub>
      <v-col-stub tag=\\"div\\" class=\\"font\\">
        Video Yardım
      </v-col-stub>
      <v-col-stub tag=\\"div\\" class=\\"font\\">
        Havale/EFT Bilgileri
      </v-col-stub>
      <v-col-stub tag=\\"div\\" class=\\"font\\">
        Müşteri Hizmetleri
      </v-col-stub>
      <v-col-stub tag=\\"div\\" class=\\"font\\">
        Ürün İadesi
      </v-col-stub>
    </v-col-stub>
    <v-col-stub tag=\\"div\\">
      Popüler Sayfalar:
      <v-col-stub tag=\\"div\\" class=\\"font\\">
        Çok Satan Kitaplar
      </v-col-stub>
      <v-col-stub tag=\\"div\\" class=\\"font\\">
        Yeni Çıkan Kitaplar
      </v-col-stub>
      <v-col-stub tag=\\"div\\" class=\\"font\\">
        Kargo Ödeyen Avantajlı Ürünler
      </v-col-stub>
      <v-col-stub tag=\\"div\\" class=\\"font\\">
        Kampanyalar
      </v-col-stub>
      <v-col-stub tag=\\"div\\" class=\\"font\\">
        Ayın Yazarları
      </v-col-stub>
      <v-col-stub tag=\\"div\\" class=\\"font\\">
        Ayın Yayınevleri
      </v-col-stub>
      <v-col-stub tag=\\"div\\" class=\\"font\\">
        Puan Kataloğu
      </v-col-stub>
      <v-col-stub tag=\\"div\\" class=\\"font\\">
        E-Dergi
      </v-col-stub>
      <v-col-stub tag=\\"div\\" class=\\"font\\">
        Yazarlar Ne okuyor?
      </v-col-stub>
      <v-col-stub tag=\\"div\\" class=\\"font\\">
        Ne okusam
      </v-col-stub>
    </v-col-stub>
    <v-col-stub tag=\\"div\\">
      Özel Sayfalar:
      <v-col-stub tag=\\"div\\" class=\\"font\\">
        Sınav Kitapları
      </v-col-stub>
      <v-col-stub tag=\\"div\\" class=\\"font\\">
        Çocuk Kitapları
      </v-col-stub>
      <v-col-stub tag=\\"div\\" class=\\"font\\">
        Cep Boy Kitaplar
      </v-col-stub>
      <v-col-stub tag=\\"div\\" class=\\"font\\">
        İngilizce Kitaplar
      </v-col-stub>
      <v-col-stub tag=\\"div\\" class=\\"font\\">
        Lisanslı Ürünler
      </v-col-stub>
      <v-col-stub tag=\\"div\\" class=\\"font\\">
        Bookinzi Kültür-Sanat Ürünleri
      </v-col-stub>
      <v-col-stub tag=\\"div\\" class=\\"font\\">
        Oyundan Daha Fazlası
      </v-col-stub>
      <v-col-stub tag=\\"div\\" class=\\"font\\">
        Montessori Oyuncakları
      </v-col-stub>
      <v-col-stub tag=\\"div\\" class=\\"font\\">
        Ahşap Puzzle
      </v-col-stub>
      <v-col-stub tag=\\"div\\" class=\\"font\\">
        Akıl Defteri
      </v-col-stub>
    </v-col-stub>
    <v-col-stub tag=\\"div\\">
      Kitapyurdu Platformları:
      <v-col-stub tag=\\"div\\" class=\\"font\\">
        Kitapyurdu Sesli Kitap
      </v-col-stub>
      <v-col-stub tag=\\"div\\" class=\\"font\\">
        KDY Platformu
      </v-col-stub>
      <v-col-stub tag=\\"div\\" class=\\"font\\">
        KDD Platformu
      </v-col-stub>
      <v-col-stub tag=\\"div\\" class=\\"font\\">
        Kütüphanem Uygulaması
      </v-col-stub>
      <v-col-stub tag=\\"div\\" class=\\"font\\">
        Kitap Kurdu Eklentisi
      </v-col-stub>
    </v-col-stub>
  </v-row-stub>
</v-card-stub>"
`;
